### PR TITLE
Deprecate oauth_scopes in flyte-cli setup-config in favor of new scopes field 

### PR DESCRIPTION
**Pull request by eapolinario**
# TL;DR
Configuration generated via `flytekit setup-config` should use the emit the credentials scopes under a new field called `scopes`. 
## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
We create a new field called `scopes` in the `flyte-cli` config file and deprecate the old one (i.e. `oauth_scopes`), while at the same time fixing the typo that prevented valid configurations from being generated.
We also ensure that we stringify all values before passing them to `ConfigParser`. 

## Tracking Issue
https://github.com/lyft/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

**Review by wild-endeavor — APPROVED** [review]

File: flytekit/clients/raw.py [hunk 1]
```diff
@@ -47,11 +47,11 @@ def _get_basic_flow_scopes() -> List[str]:
     deprecated_single_scope = _DEPRECATED_SCOPE.get()
     if deprecated_single_scope:
         return [deprecated_single_scope]
-    oauth_scopes = OAUTH_SCOPES.get()
-    if "openid" in oauth_scopes:
+    scopes = SCOPES.get()
```
Comment (wild-endeavor) [inline flytekit/clients/raw.py:50]: i think we may have to do something uglier like

```suggestion
    scopes = SCOPES.get() or OAUTH_SCOPES.get()
    if scopes and "openid" in scopes:
```

**Review by wild-endeavor — APPROVED** [review]
